src/TCommandLine.cpp

```diff
@@ -1020,7 +1014,11 @@ void TCommandLine::historyMove(MoveDirection direction)
             mHistoryBuffer = 0;
         }
         setPlainText(mHistoryList[mHistoryBuffer]);
-        selectAll();
+        if (mpHost->mHighlightHistory) {
+            selectAll();
+        } else {
+            moveCursor(QTextCursor::End);
+        }
```

Comment:
> #### Brief overview of PR changes/additions
> There was a bunch of duplicated code after the `enterCommand()` calls in TCommandLine.
> 
> This PR has three commits, which may be easier to review separately:
> 
> * The first removes the lines that were useless; they duplicated lines already done in `enterCommand()`.
> * The second moves the lines that appear after both calls to `enterCommand()` inside the function call
> * The third is like the second, but for `historyMove`.

Indeed reviewing the 3 commits seperately was enabling even little old me to see what happened here.
Commit 1 and 2 seem legit and well done! 
Commit 3 I can't quite grasp as a whole, yet. 
Not only was code moved like 2, but `selectAll();` incorporated in the moved `if (mpHost->mHighlightHistory)` clause.
Not sure if this works, too tired to think it through, also did not yet try the build and search changes in behavior.
Maybe you can elaborate why this is reasonable. Maybe others will understand immediately as well.